### ISSUE TITLE
Stack Overflow and PortEnterCriticalTimeout in NMEA Parser Example Code (IDFGH-9707)

### ISSUE THREAD

**Issue by violinoleum** [espressif/esp-idf#11044]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### General issue report

Board: ESP32-S3 WROOM-2
ESP-IDF Version: 5.0
Using Espressif VS Code Extension
GPS Model: Neo-6M GT-U7

Hello, I made a project template using the NMEA Parser Example, and I ran the code but I'm running into a stack overflow issue and then a Port Critical Timeout Issue. My log file is pasted below. Do you know what could be causing the issue? I turned off statement support for all except GLL, but the same issue occurs. I also increased my NMEA Parser Ring Buffer Size from 1024 to 2048, and there is still the same issue. I also tried the "Detect flash size when flashing bootloader" option in the configuration, but it also didn't fix the issue;

I (209) cpu_start: Application information:
I (211) cpu_start: Project name:     GPSTest
I (216) cpu_start: App version:      1
I (221) cpu_start: Compile time:     Mar 23 2023 15:25:51
I (227) cpu_start: ELF file SHA256:  50dd0e75474fda06...
I (233) cpu_start: ESP-IDF:          v5.0.1
I (238) cpu_start: Min chip rev:     v0.0
I (242) cpu_start: Max chip rev:     v0.99 
I (247) cpu_start: Chip rev:         v0.1
I (252) heap_init: Initializing. RAM available for dynamic allocation:
I (259) heap_init: At 3FC95078 len 00054698 (337 KiB): D/IRAM
I (266) heap_init: At 3FCE9710 len 00005724 (21 KiB): STACK/DRAM
I (272) heap_init: At 3FCF0000 len 00008000 (32 KiB): DRAM
I (278) heap_init: At 600FE010 len 00001FF0 (7 KiB): RTCRAM
I (286) spi_flash: detected chip: mxic
W (289) spi_flash: Detected flash size > 16 MB, but access beyond 16 MB is not supported for this flash model yet.
I (300) spi_flash: flash io: dio
I (305) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (325) uart: queue free spaces: 16
I (325) nmea_parser: NMEA Parser init OK
W (345) gps_demo: Unknown statement:PGLL,,,,,,V,N*64

W (1215) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (1265) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (1315) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (1365) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (1415) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (2215) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (2265) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (2315) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (2365) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (2415) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (3215) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (3265) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (3315) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (3365) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (3415) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (4215) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (4265) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (4315) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (4365) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (4415) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (5215) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (5265) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (5315) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (5365) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (5415) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (6215) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

***ERROR*** A stack overflow in task nmea_parser has been detected.

Backtrace: 0x40375d8a:0x3fceb330 0x4037aae1:0x3fceb350 0x4037dba6:0x3fceb370 0x4037c94b:0x3fceb3f0 0x4037ab98:0x3fceb410 0x4037ab8e:0x3fceb460 0x4201dae5:0x000000fd |<-CORRUPTED
0x40375d8a: panic_abort at C:/Users/User/esp/esp-idf/components/esp_system/panic.c:423

0x4037aae1: esp_system_abort at C:/Users/User/esp/esp-idf/components/esp_system/esp_system.c:153

0x4037dba6: vApplicationStackOverflowHook at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:391

0x4037c94b: vTaskSwitchContext at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/tasks.c:3648

0x4037ab98: _frxt_dispatch at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/portasm.S:418

0x4037ab8e: _frxt_int_exit at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/portasm.S:213

0x4201dae5: handler_execute at C:/Users/User/esp/esp-idf/components/esp_event/esp_event.c:137

ELF file SHA256: 50dd0e75474fda06

Rebooting...
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x3 (RTC_SW_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40375729
0x40375729: esp_restart_noos_dig at C:/Users/User/esp/esp-idf/components/esp_system/esp_system.c:64 (discriminator 1)

SPIWP:0xee
Octal Flash Mode Enabled
For OPI Flash, Use Default Flash Boot Mode
mode:SLOW_RD, clock div:1
load:0x3fce3810,len:0x1648
load:0x403c9700,len:0xbe0
load:0x403cc700,len:0x2ef8
SHA-256 comparison failed:
Calculated: 960621295dc2eb45da20c08aeb735100d3eaae6c9b78889863b70d7a07fbfbc6
Expected: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
Attempting to boot anyway...
entry 0x403c9900
I (55) boot: ESP-IDF v5.0.1 2nd stage bootloader
I (55) boot: compile time 15:26:06
I (55) boot: chip revision: v0.1
I (57) boot.esp32s3: Boot SPI Speed : 80MHz
I (62) boot.esp32s3: SPI Mode       : SLOW READ
I (67) boot.esp32s3: SPI Flash Size : 32MB
I (72) boot: Enabling RNG early entropy source...
I (77) boot: Partition Table:
I (81) boot: ## Label            Usage          Type ST Offset   Length
I (88) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (95) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (103) boot:  2 factory          factory app      00 00 00010000 00100000
I (110) boot: End of partition table
I (115) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=0a61ch ( 42524) map
I (130) esp_image: segment 1: paddr=0001a644 vaddr=3fc92300 size=02348h (  9032) load
I (133) esp_image: segment 2: paddr=0001c994 vaddr=40374000 size=03684h ( 13956) load
I (143) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=1ea1ch (125468) map
I (168) esp_image: segment 4: paddr=0003ea44 vaddr=40377684 size=0aba0h ( 43936) load
I (182) boot: Loaded app from partition at offset 0x10000
I (182) boot: Disabling RNG early entropy source...
I (193) cpu_start: Pro cpu up.
I (193) cpu_start: Starting app cpu, entry point is 0x4037526c
0x4037526c: call_start_cpu1 at C:/Users/User/esp/esp-idf/components/esp_system/port/cpu_start.c:142

I (0) cpu_start: App cpu up.
I (208) cpu_start: Pro cpu start user code
I (208) cpu_start: cpu freq: 160000000 Hz
I (209) cpu_start: Application information:
I (211) cpu_start: Project name:     GPSTest
I (216) cpu_start: App version:      1
I (221) cpu_start: Compile time:     Mar 23 2023 15:25:51
I (227) cpu_start: ELF file SHA256:  50dd0e75474fda06...
I (233) cpu_start: ESP-IDF:          v5.0.1
I (238) cpu_start: Min chip rev:     v0.0
I (242) cpu_start: Max chip rev:     v0.99 
I (247) cpu_start: Chip rev:         v0.1
I (252) heap_init: Initializing. RAM available for dynamic allocation:
I (259) heap_init: At 3FC95078 len 00054698 (337 KiB): D/IRAM
I (266) heap_init: At 3FCE9710 len 00005724 (21 KiB): STACK/DRAM
I (272) heap_init: At 3FCF0000 len 00008000 (32 KiB): DRAM
I (278) heap_init: At 600FE010 len 00001FF0 (7 KiB): RTCRAM
I (286) spi_flash: detected chip: mxic
W (289) spi_flash: Detected flash size > 16 MB, but access beyond 16 MB is not supported for this flash model yet.
I (300) spi_flash: flash io: dio
I (305) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (325) uart: queue free spaces: 16
I (325) nmea_parser: NMEA Parser init OK
W (925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (1025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (1075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (1125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (1925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (1975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (2025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (2075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (2125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (2925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (2975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (3025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (3075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (3125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (3925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (3975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (4025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (4075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (4125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (4925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (4975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (5025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (5075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (5125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (5925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (5975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (6025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (6075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (6125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (6925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (6975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (7025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (7075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (7125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (7925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (7975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (8025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (8075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (8125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (8925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (8975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (9025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (9075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (9125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
W (9925) gps_demo: Unknown statement:$GPRMC,,V,,,,,,,,,,N*53

W (9975) gps_demo: Unknown statement:$GPVTG,,,,,,,,,N*30

W (10025) gps_demo: Unknown statement:$GPGGA,,,,,,0,00,99.99,,,,,,*48

W (10075) gps_demo: Unknown statement:$GPGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*30

I (10125) gps_demo: 2000/0/0 63:87:52 => 
                                                latitude   = 0.00000°N
                                                longitude = 0.00000°E
                                                altitude   = 0.00m
                                                speed      = 0.000000m/s
Guru Meditation Error: Core  0 panic'ed (Interrupt wdt timeout on CPU0). 

Core  0 register dump:
PC      : 0x4037da0f  PS      : 0x00060f34  A0      : 0x8037ba48  A1      : 0x3fcf35b0
0x4037da0f: spinlock_acquire at C:/Users/User/esp/esp-idf/components/esp_hw_support/include/spinlock.h:112
 (inlined by) xPortEnterCriticalTimeout at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:311

A2      : 0x3fceaee0  A3      : 0xffffffff  A4      : 0xb33fffff  A5      : 0x00060f23
A6      : 0x625fd8d0  A7      : 0x0000cdcd  A8      : 0x8037da15  A9      : 0x3fcf3590
A10     : 0x00000000  A11     : 0xb33fffff  A12     : 0x0000cdcd  A13     : 0x00000000
A14     : 0x00000001  A15     : 0x00000000  SAR     : 0x00000016  EXCCAUSE: 0x00000005
EXCVADDR: 0x00000000  LBEG    : 0x400570e8  LEND    : 0x400570f3  LCOUNT  : 0xffffffff  

Backtrace: 0x4037da0c:0x3fcf35b0 0x4037ba45:0x3fcf35e0 0x4037bba0:0x3fcf3620 0x4201d906:0x3fcf3640 0x4201d986:0x3fcf3670 0x42008612:0x3fcf36a0 0x42007fad:0x3fcf36c0 0x4201e37b:0x3fcf3700 0x4037d861:0x3fcf3730
0x4037da0c: spinlock_acquire at C:/Users/User/esp/esp-idf/components/esp_hw_support/include/spinlock.h:112
 (inlined by) xPortEnterCriticalTimeout at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:311

0x4037ba45: vPortEnterCritical at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include/freertos/portmacro.h:573
 (inlined by) xQueueSemaphoreTake at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/queue.c:1570

0x4037bba0: xQueueTakeMutexRecursive at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/queue.c:738

0x4201d906: esp_event_handler_unregister_with_internal at C:/Users/User/esp/esp-idf/components/esp_event/esp_event.c:780

0x4201d986: esp_event_handler_unregister_with at C:/Users/User/esp/esp-idf/components/esp_event/esp_event.c:806

0x42008612: nmea_parser_remove_handler at C:/Users/User/GPSTest/main/nmea_parser.c:785

0x42007fad: app_main at C:/Users/User/GPSTest/main/nmea_parser_example_main.c:67

0x4201e37b: main_task at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/port_common.c:131 (discriminator 2)

0x4037d861: vPortTaskWrapper at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:154

Core  1 register dump:
PC      : 0x40378b46  PS      : 0x00060834  A0      : 0x82002469  A1      : 0x3fcf4590
0x40378b46: esp_cpu_wait_for_intr at C:/Users/User/esp/esp-idf/components/esp_hw_support/cpu.c:110

A2      : 0x00000000  A3      : 0x0000abab  A4      : 0x8037db5e  A5      : 0x3fcf1360
A6      : 0x00060023  A7      : 0x00000000  A8      : 0x8200c0d6  A9      : 0x3fcf4550
A10     : 0x00000000  A11     : 0x00000003  A12     : 0x00000008  A13     : 0x00000101
A14     : 0x0000000c  A15     : 0xfffffff6  SAR     : 0x00000000  EXCCAUSE: 0x00000005
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000

Backtrace: 0x40378b43:0x3fcf4590 0x42002466:0x3fcf45b0 0x4037c181:0x3fcf45d0 0x4037d861:0x3fcf45f0
0x40378b43: xt_utils_wait_for_intr at C:/Users/User/esp/esp-idf/components/xtensa/include/xt_utils.h:81
 (inlined by) esp_cpu_wait_for_intr at C:/Users/User/esp/esp-idf/components/esp_hw_support/cpu.c:101

0x42002466: esp_vApplicationIdleHook at C:/Users/User/esp/esp-idf/components/esp_system/freertos_hooks.c:59

0x4037c181: prvIdleTask at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/tasks.c:4157 (discriminator 1)

0x4037d861: vPortTaskWrapper at C:/Users/User/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:154

ELF file SHA256: 50dd0e75474fda06

Rebooting...
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x3 (RTC_SW_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40375729
0x40375729: esp_restart_noos_dig at C:/Users/User/esp/esp-idf/components/esp_system/esp_system.c:64 (discriminator 1)

**Comment by ESP-Marius** [comment]
Did you try increasing `CONFIG_NMEA_PARSER_TASK_STACK_SIZE` in menuconfig? The default of 2048 bytes does seem a bit small..

**Comment by violinoleum** [comment]
Thank you for your advice. I did try to increase the stack size, but it still crashed. It seems that there is also a limit to how large the stack can be, according to the code in Kconfig.projbuild:

config NMEA_PARSER_TASK_STACK_SIZE
        int "NMEA Parser Task Stack Size"
        range 0 4096
        default 2048
        help
            Stack size of NMEA Parser task.

The max value is 4096 if I'm interpreting it correctly.

Also I did some more debugging, and it seems that the issue only happens after 10 seconds, once the code reaches the nmea_parser_remove_handler() and nmea_parser_deinit() in app_main(). So I "fixed" the issue by replacing 
vTaskDelay(10000 / portTICK_PERIOD_MS); in app_main() with 

while(1){
        vTaskDelay(1000 / portTICK_PERIOD_MS);
 }

so that it does not reach the nmea_parser_remove_handler(). This likely does not solve the underlying issue, but it is enough for the purpose that I'm using the code for.

Thank you for your help with resolving this issue!

**Comment by ESP-Marius** [comment]
I just did a quick test locally, the example does indeed have the potential for issues if ran with the default stack size of 2048 as we are right on the stack border.

I'll increase the default value.

Are you sure you are still having issues if you change the size to 4000 (and no other changes to the example)? At least it seems to work fine for me.

**Comment by violinoleum** [comment]
Hi, I followed your advice and changed the stack size to 4000, and it fixed the issue. I think when I thought I changed the stack size before, I had really just changed the NMEA Parser Ring Buffer Size instead, and I misremembered. With a stack size of 4000, it consistently works now, even when testing the vTaskDelay as 30 seconds instead of the original 10 seconds. Thank you so much for your help!
